### PR TITLE
[OCPQE-26995] add vsphere disconnected support for winc

### PR DESCRIPTION
**Pull request by weinliu**
/hold

### CODE REVIEW

**Comment by weinliu** [comment]
/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected



<br class="Apple-interchange-newline">

**Comment by weinliu** [comment]
/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort


/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

File: ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh [hunk 1]
```diff
@@ -187,9 +205,21 @@ oc wait nodes -l kubernetes.io/os=windows --for condition=Ready=True --timeout=1
 # Determine Windows container version based on OS version
 os_version=$(oc get nodes -l 'kubernetes.io/os=windows' -o=jsonpath="{.items[0].status.nodeInfo.osImage}")
 
-windows_container_image="mcr.microsoft.com/powershell:lts-nanoserver-ltsc2022"
-if [[ "$os_version" == *"2019"* ]]; then
-    windows_container_image="mcr.microsoft.com/powershell:lts-nanoserver-1809"
+# Setup container image paths based on environment
+if [ -n "${DISCONNECTED_IMAGE_REGISTRY:-}" ]; then
+    # Use disconnected registry paths
+    windows_container_image="${DISCONNECTED_IMAGE_REGISTRY}/powershell:lts-nanoserver-ltsc2022"
```
Comment (rrasouli) [inline ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh:211]: vSphere supports ONLY server2022, this code is redundant

File: ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh [hunk 2]
```diff
@@ -4,26 +4,74 @@ set -o nounset
...
+    oc get events -n winc-test --sort-by='.lastTimestamp' | grep $deployment_name
+}
 
-  # Display pods and configmap
-  oc get pod -owide -n winc-test
-  oc get cm winc-test-config -oyaml -n winc-test
+# Function to handle disconnected environment setup
+function disconnected_prepare() {
+    # Extract registry hostname from image
+    export DISCONNECTED_REGISTRY=$(echo $1 | awk -F/ '{print $1}')
```
Comment (rrasouli) [inline ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh:39]: Try this:
```
DISCONNECTED_REGISTRY=$(echo "$1" | awk -F/ '{print $1}')
export DISCONNECTED_REGISTRY
```

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

File: ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh [hunk 3]
```diff
@@ -1,29 +1,102 @@
-#!/bin/bash
+#!bin/bash
```
Comment (rrasouli) [inline ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh:1]: Wrong path `#!/bin/bash`

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnecte

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnecte

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

File: ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh [hunk 4]
```diff
@@ -111,40 +258,79 @@ spec:
...
 }
 
-IAAS_PLATFORM=$(oc get infrastructure cluster -o=jsonpath="{.status.platformStatus.type}"| tr '[:upper:]' '[:lower:]')
+# Main script execution starts here
 
+# Get infrastructure platform type
+IAAS_PLATFORM=$(oc get infrastructure cluster -o=jsonpath="{.status.platformStatus.type}"| tr '[:upper:]' '[:lower:]')
 
+# Get Windows machineset information
 winworker_machineset_name=$(oc get machineset -n openshift-machine-api -o json | jq -r '.items[] | select(.metadata.name | test("win")).metadata.name')
```
Comment (rrasouli) [inline ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh:301]: `export winworker_machineset_replicas=$(oc get machineset -n openshift-machine-api $winworker_machineset_name -o jsonpath="{.spec.replicas}")`
To fix shelshock issues

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by rrasouli** [comment]
That is very odd, I don't see the file `.//ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh` nor the line `oc get machineset -n openshift-machine-api $winworker_machineset_name` that the shellcheck is complaining about

**Comment by weinliu** [comment]
> That is very odd, I don't see the file `.//ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh` nor the line `oc get machineset -n openshift-machine-api $winworker_machineset_name` that the shellcheck is complaining about

@rrasouli I still have cleaning work to do before it's ready for review

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
/pj-rehearse abort

/pj-rehearse pull-ci-openshift-openshift-tests-private-release-4.16-debug-winc-vsphere-ipi-disconnected

**Comment by weinliu** [comment]
rebasing with https://github.com/openshift/release/pull/59557

**Pull request closed without merging**